src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -40,7 +40,7 @@ struct SemidiscretizationHyperbolic{Mesh, Equations, InitialCondition,
                                                                  BoundaryConditions,
                                                                  SourceTerms, Solver,
                                                                  Cache}
-        @assert ndims(mesh) == ndims(equations)
+        #@assert ndims(mesh) == ndims(equations)
```

Comment:
I removed this check because the new implementation requires a 2D mesh (`P4estMesh{2}`) with a 3D equation. Should I just remove this line or maybe change the check?

Reply:
That's a good question. I think we should analyze where we use this kind of assumption and try to get it right. For example, I guess everything will fail badly if you use an `UnstructuredMesh2D` with a 3D equation, won't it?

Reply:
Yes, it will fail. We would need special constructors that create those meshes using nodes with three (x, y, z) node coordinates, and we would then need to adjust the metric term computation of all mesh types. It is doable but I think it is out of the scope of the present PR and probably no one will ever use it :rofl:. 

Then maybe I just check for the particular case of a `P4estMesh2D` with 3D node coordinates in combination with a 3D equation, allow that combination and bring back the `@assert` for any other case. Does that sound like a good solution?